source/agora/consensus/EnrollmentManager.d

```diff
@@ -471,6 +471,33 @@ public class EnrollmentManager
         return this.validator_set.hasPreimage(enroll_key, distance);
     }
 
+    /***************************************************************************
+
+        Generate the random seed reduced from the preimages for the provided
+        block height.
+
+        Params:
+            keys = the keys to look up
+            height = the desired block height to look up the images in
+
+        Returns:
+            the random seed
+
+    ***************************************************************************/
+
+    public Hash getRandomSeed (in Hash[] keys, in Height height) @safe nothrow
```

Comment:
Forgot to write a unittest here. There is a networking unittest, but I think this function should be introduced in its own commit with a unittest.